BarrysPeanutsTemporalJava/src/main/java/barryspeanuts/App.java

```diff
@@ -0,0 +1,127 @@
+package barryspeanuts;
+
+import barryspeanuts.model.*;
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowOptions;
+import io.temporal.common.RetryOptions;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class App {
+  static final String TASK_QUEUE = "BarryPeanutsTemporal";
+  static final String WORKFLOW_ID = TASK_QUEUE + "-" + UUID.randomUUID();
+  private static final Logger logger = LoggerFactory.getLogger(App.class);
+
+  @SuppressWarnings("CatchAndPrintStackTrace")
+  public static void main(String[] args) {
+    // gRPC stubs wrapper that talks to the local docker instance of temporal
+    // service.
+    WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
+    // client that can be used to start and signal workflows
+    WorkflowClient client = WorkflowClient.newInstance(service);
+
+    // worker factory that can be used to create workers for specific task queues
+    WorkerFactory factory = WorkerFactory.newInstance(client);
+
+    // Worker that listens on a task queue and hosts both workflow and activity
+    // implementations.
+    Worker worker = factory.newWorker(TASK_QUEUE);
+
+    // Workflows are stateful. So you need a type to create instances.
+    worker.registerWorkflowImplementationTypes(ShoppingCartWorkflowImpl.class);
+
+    // Start all workers created by this factory.
+    factory.start();
+    logger.info("Worker started for task queue: {} with WorkflowID : {}.", TASK_QUEUE, WORKFLOW_ID);
+
+    // now we can start running instances of our workflow - its state will be persisted
```

Comment:
I'm wondering if initializing the worker and running the demo both in `main()` may not be giving users that this is way more complicated than it really is.

I've seen a few people asking questions make it clear that they were under the impression that they had to run one worker per workflow that they were launching, and I'm afraid that this example could actually lead to that kind of misunderstanding.

Reply:
@mjameswh happy to do so if there is consensus between you and @cretz  that this is something I should do. Presently I am working of the style I've seen in many of the examples published by Temporal.

Reply:
@cretz and @mjameswh is this a showstopper I need to address?

Reply:
Not a showstopper

Reply:
OK, as you know, I've been having group isolating the worker and the workflow into separate processes. So, for now I'll assume we can go with this. However, it is on my list to resolve.